### DISCUSSION TITLE
failed to load 'plugin.py'

### DISCUSSION THREAD

**Discussion by domoticauser** [addiejanssen/domoticz-solaredge-modbustcp-plugin#61] [Q&A]
Hi,

I have this error, what to do?

 2023-03-11 18:16:32.341 Error: Solaredge: (SolarEdge_ModbusTCP) failed to load 'plugin.py', Python Path used was '/home/pi/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/:/usr/lib/python39.zip:/usr/lib/python3.9:/usr/lib/python3.9/lib-dynload:/usr/local/lib/python3.9/dist-packages:/usr/lib/python3/dist-packages:/usr/lib/python3.9/dist-packages'.
2023-03-11 18:16:32.344 Error: Solaredge: Traceback (most recent call last):
2023-03-11 18:16:32.344 Error: Solaredge: File "/home/pi/domoticz/plugins/domoticz-solaredge-modbustcp-plugin/plugin.py", line 54, in <module>
2023-03-11 18:16:32.344 Error: Solaredge: import solaredge_modbus
2023-03-11 18:16:32.344 Error: Solaredge: File "/usr/local/lib/python3.9/dist-packages/solaredge_modbus/__init__.py", line 7, in <module>

**Comment by domoticauser** [comment]
sorry, already a topic.

**Comment by Xenomes** [comment]
Did you install `sudo pip3 install -r requirements.txt`?

**Reply by domoticauser** [reply]
Yes i did, to be sure, did it again.
It looks the same as the other topic. The only difference is the phyton version.

 pi@domoticzVM:~/domoticz/plugins/domoticz-solaredge-modbustcp-plugin$ sudo pip3 install -r requirements.txt                                                     [sudo] wachtwoord voor pi:
Requirement already satisfied: solaredge_modbus==0.7.0 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 1)                             ) (0.7.0)
Requirement already satisfied: pymodbus>=2.4.0 in /usr/local/lib/python3.9/dist-packages (from solaredge_modbus==0.7.0->-r requirem                             ents.txt (line 1)) (3.1.3)


It works on my rpi, i want to replace it with a new system.
Version: 2023.1
Build Hash: f9b9ac774
Compile Date: 2023-02-14 15:06:40
dzVents Version: 3.1.8
Python Version: 3.7.3 (default, Oct 31 2022, 14:04:00) [GCC 8.3.0]

And not on this system, here i get the error.
Version: 2023.1
Build Hash: f9b9ac774
Compile Date: 2023-02-14 15:06:40
dzVents Version: 3.1.8
Python Version: 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]

**Comment by Xenomes** [comment]
Yes i see, Python 3.9
Maybe this works `sudo pip3 install solaredge_modbus -U` and otherwise without the sudo.

**Comment by domoticauser** [comment]
Hi,
Thanks for your support!

So far the error is gone... I think it works. But it is dark at this moment ;), so that could be the problem at this moment.
2023-03-11 21:30:55.119 Solaredge: Connection established with: 192.168.1.101:11502 Device Address: 1. BUT... inverter returned no information

It only looks that domoticz crashes, when i change anything. After restart it works again.


But i see 2 different versions of the plugin.

The old system, (working)
![afbeelding](https://user-images.githubusercontent.com/11601535/224510185-1f4387e8-8c00-4300-b8a1-d739634c53d4.png)


My new system (with the error)  Need to test with daylight
![afbeelding](https://user-images.githubusercontent.com/11601535/224510134-460d085d-6593-41d4-baf5-01c56fb7698e.png)

**Comment by Xenomes** [comment]
Strange the first screenshot is showing two plugins in one, that can happen if there are two plugin with the same internal name.
What command was the solution with or without sudo?
Otherwise run `sudo pip3 install solaredge_modbus==0.7.0` to force it back to 0.7.0 also with or without sudo.

**Reply by domoticauser** [reply]
The two plugins in one screen looks like a browser cache problem. Is solved! It was my working system.

I will let you know if my new system is working with the plugin, for that we need daylight The error is gone!

**Comment by domoticauser** [comment]
Yes it works! Keep up the good work.

Thanks for your support.
